core/src/utils/animation/animation-utils.ts

```diff
@@ -1,10 +1,28 @@
+let animationPrefix: string | null = null;
+
+export const getAnimationPrefix = (ele: HTMLElement): string => {
+  if (animationPrefix === null) {
+    const eleStyle: { [key: string]: any } = ele.style;
+    animationPrefix = '';
+    if (eleStyle.animationName === undefined) {
+      const domPrefixes = 'Webkit Moz O ms Khtml'.split(' ');
+      for (const prefix of domPrefixes) {
+        if (eleStyle[prefix + 'AnimationName'] !== undefined) {
+          animationPrefix = '-' + prefix.toLowerCase() + '-';
+          break;
+        }
+      }
+    }
+  }
+  return animationPrefix;
+};
 
 export const setStyleProperty = (element: HTMLElement, propertyName: string, value: string | null) => {
-  element.style.setProperty(propertyName, value);
+  element.style.setProperty(getAnimationPrefix(element) + propertyName, value);
```

Comment:
Since the `setStyleProperty` and `removeStyleProperty` functions are used for other things not related to animation props, I think it makes sense to only apply the prefix if `propertyName.includes('animation')`.

Reply:
Ok that's right. I will use `startsWith`

Reply:
Typescript do not allow me to test the `propertyName` to be undefined or null without any cast. Therefore I did not test for an `undefined` or `null` on the `propertyName`.  Typescript prevent that the `propertyName` can be null, but I do not know this is true for any usecase in a javascript environment. So my question is should I add an null/undefined check for the `propertyName`?

Reply:
You can assume that `propertyName` is always a string and never undefined.